### ISSUE TITLE
Create _gen variant for //protoc-gen-go/generator

### ISSUE THREAD

**Issue by drigz** [bazelbuild/rules_go#1817]
The target `@com_github_golang_protobuf//protoc-gen-go/generator:go_default_library` depends on `@com_github_golang_protobuf//protoc-gen-go/descriptor:go_default_library`. This can lead to a conflict:

```
GoLink: warning: package "github.com/golang/protobuf/protoc-gen-go/descriptor" is provided by more than one rule:
    @com_github_golang_protobuf//protoc-gen-go/descriptor:go_default_library
    @io_bazel_rules_go//proto/wkt:descriptor_go_proto
Set "importmap" to different paths in each library.
This will be an error in the future.
```

This warning occurs when compiling grpc-gateway with the latest rules_go/gazelle, as [types.go](https://github.com/grpc-ecosystem/grpc-gateway/blob/7d429ea4af6c17ae773bafb6dffae8e851a9b35f/protoc-gen-grpc-gateway/descriptor/types.go#L7) depends on both `protoc-gen-go/descriptor` and `protoc-gen-go/generator`.

I would expect the target `@com_github_golang_protobuf//protoc-gen-go/generator:go_default_library_gen` to exist to avoid this problem. Could this be added?
